crates/tauri/src/scope/fs.rs

```diff
@@ -470,4 +472,79 @@ mod tests {
       assert!(scope.is_allowed("C:\\home\\tauri\\anyfile"));
     }
   }
+
+  #[cfg(windows)]
+  #[test]
+  fn windows_root_paths() {
+    let scope = new_scope();
+    {
+      // UNC network path
+      scope.allow_directory("\\\\localhost\\c$", true).unwrap();
+      assert!(scope.is_allowed("\\\\localhost\\c$"));
+      assert!(scope.is_allowed("\\\\localhost\\c$\\Windows"));
+      // Does not work because push_pattern adds a trailing backslash to the base
+      // pattern before "\**" is added leading to a pattern of "\\localhost\c$\\**"
+      // which does not match.
+      // assert!(scope.is_allowed("\\\\localhost\\c$\\NonExistentFile"));
+      assert!(!scope.is_allowed("\\\\localhost\\d$"));
+      assert!(!scope.is_allowed("\\\\OtherServer\\Share"));
+    }
+
+    let scope = new_scope();
+    {
+      // Verbatim UNC network path
+      scope
+        .allow_directory("\\\\?\\UNC\\localhost\\c$", true)
+        .unwrap();
+      assert!(scope.is_allowed("\\\\localhost\\c$"));
+      assert!(scope.is_allowed("\\\\localhost\\c$\\Windows"));
+      // Does not work because non-existent files can't be canonicalized, therefore
+      // the verbatim prefix ("\\?\UNC\") is not added.
+      // assert!(scope.is_allowed("\\\\localhost\\c$\\Windows\\NonExistentFile"));
```

Comment:
Not sure I understand why this won't work, I see that `"\\\\localhost\\c$\\Windows"` works but why not `"\\\\localhost\\c$\\Windows\\NonExistentFile"`, isn't it just one level deeper?

Reply:
The problem is if you allow a directory with a verbatim path (e.g. \\?\UNC\localhost\c$), then the non-verbatim version of the path (e.g. \\localhost\c$) is not allowed. In general, I'm not sure it's safe to convert a verbatim path to a non-verbatim path so this is probably desirable behavior but it is a bit surprising.

Reply:
Note that `dunce::simplified()` doesn't simplify UNC network paths, only disk based paths.

Reply:
I would just ignore this case then and leave it up to the user to specify both versions of the path